### DISCUSSION TITLE
Preselected today's date in my DateRangeFilter (unwanted)

### DISCUSSION THREAD

**Discussion by Psenicinii-Valerian** [malzariey/filament-daterangepicker-filter#126] [General]
When I just open to check my DateRangeFilter it automatically sets me a default today's date - which is unwanted for me at this moment. The thing is that even tho I do not click on "Apply" button, if I leave the modal (where choosing a date) it sets me automatically as filter today's date. Same thing occurs with "Cancel" button. In every case it keeps me as active filter today's date. Currently the only way to avoid this unwanted "feature" is to select other date(range of dates). But what if I accidentally click on my DateRangePicker? I always have to clear the filter as it sets automatically for today's date? Maybe anyone has an idea on how to solve this? (*I do not have enabled the defaultToday() method). This is how my DateRangePicker Component looks like:
 DateRangeFilter::make('date')
                    ->label('Data')
                    ->disableClear()
                    ->separator(' - ')
                    ->linkedCalendars(false)
                    ->disableRanges()
                    ->placeholder("Enter Data")
                    ->showDropdowns()->minYear(2010)->maxYear(2030)
                    ->startDate(null)
                    ->endDate(null),
In the attachments below you can see how my DateRangePicker displays when clicking (it has preselected today's date - with blue bacground)
![image](https://github.com/user-attachments/assets/2dde041a-fa99-4631-b40f-a9ea49ce92a4)
And I still get this filter everytime I try to press "Cancel" button or click outside the modal.
![image](https://github.com/user-attachments/assets/5d9415ba-8365-470c-ab82-62fd854b6225)

**Comment by malzariey** [comment]
It is not preselected 

Its just the picker has to show a date, If you pressed cancel button the fitler will not be initilized.

**Reply by Psenicinii-Valerian** [reply]
in fact I'm pressing cancel button and the filter it is initialized (as in the photo)
same occurs when I click out of the modal datepicker

**Reply by Psenicinii-Valerian** [reply]
Here is a record in order to better show what i'm talking about
https://github.com/user-attachments/assets/6329ea45-3efd-4277-9832-f88e00b9fd2f

Whenever I click on cancel or out of the DateRangePicker modal, i'd love not to start any filtering
Unfortunately, the filtering indeed happens and it doesn't really take count that I just click out of the modal or click cancel
I did not modify any files from the source code (be it in vendor or public->css/js->malzariey)
How can I fix this tho - bcs I dont want this behaviour in my app?

Again, this is the code for my 2 filter components:

```
DateRangeFilter::make('date')
    ->label('Data Facturii')
    ->disableClear()
    ->separator(' - ')
    ->linkedCalendars(false)
    ->disableRanges()
    ->placeholder("Introdu Data Facturii")
    ->showDropdowns()->minYear(2010)->maxYear(2030)
    ->startDate(null)
    ->endDate(null),
DateRangeFilter::make('due_date')
    ->label('Data Scadenta Factura')
    ->disableClear()
    ->separator(' - ')
    ->linkedCalendars(false)
    ->disableRanges()
    ->placeholder("Introdu Data Scadenta Factura")
    ->showDropdowns()->minYear(2010)->maxYear(2030)
    ->startDate(null)
    ->endDate(null),
```
(I've tried lots of methods variations on this component but always I had the same result with this auto filter selection behaviour)

**Reply by clk11** [reply]
I have the same issue. For some reason, when I cancel the selection, it defaults to the initial option.

**Comment by senaranya** [comment]
I'm facing the same issue. As a temporary workaround I had to manually clear the date if it is today. Great to have a permanent fix for this as my workaround is far from perfect:
```php
 ->afterStateUpdated(function (?string &$state, ?string $old) use ($dateRangePicker) {
    if ($old === null && $state === today()->format('m-d-Y') . ' - ' . today()->format('m-d-Y')) {
        $dateRangePicker->clear();
    }
});
```

**Comment by malzariey** [comment]
Issue has been addressed in the beta version, please test the version **3.1.beta.** 

Thanks for using the plugin
